clients/documents/documents.go

```diff
@@ -204,239 +216,194 @@ func (c Client) List(ctx context.Context, filter string) (ListResponse, error) {
 	return retVal, nil
 }
 
-func (c Client) Create(ctx context.Context, name string, isPrivate bool, externalId string, data []byte, documentType DocumentType) (Response, error) {
-	body := &bytes.Buffer{}
-	writer := multipart.NewWriter(body)
-
-	if err := writer.WriteField("type", string(documentType)); err != nil {
-		return Response{}, err
-	}
-	if err := writer.WriteField("name", name); err != nil {
-		return Response{}, err
+func (c Client) Create(ctx context.Context, name string, isPrivate bool, externalId string, data []byte, documentType DocumentType) (api.Response, error) {
+	d := documents.Document{
+		Kind:       string(documentType),
+		Name:       name,
+		Public:     !isPrivate,
+		ExternalID: externalId,
+		Content:    data,
 	}
 
-	if err := writer.WriteField("isPrivate", strconv.FormatBool(isPrivate)); err != nil {
-		return Response{}, err
+	resp, err := c.create(ctx, d)
+	if err != nil {
+		return api.Response{}, err
 	}
 
-	if externalId != "" {
-		if err := writer.WriteField("externalId", externalId); err != nil {
-			return Response{}, err
-		}
+	var md documentMetadata
+	if md, err = unmarshallMetadata(resp.Data); err != nil {
+		return api.Response{}, err
 	}
 
-	part, err := writer.CreatePart(map[string][]string{
-		"Content-Type":        {"application/json"},
-		"Content-Disposition": {fmt.Sprintf(`form-data; name="content"; filename="%s"`, name)},
-	})
+	r, err := c.patch(ctx, md.ID, md.Version, d)
 	if err != nil {
-		return Response{}, err
+		if _, err1 := c.delete(ctx, md.ID, md.Version); err1 != nil {
+			return api.Response{}, errors.Join(err, err1)
+		}
+		return api.Response{}, err
 	}
 
-	if _, err = part.Write(data); err != nil {
-		return Response{}, err
+	return r, nil
+}
+
+func (c Client) Update(ctx context.Context, id string, name string, isPrivate bool, data []byte, documentType DocumentType) (api.Response, error) {
+	if id == "" {
+		return api.Response{}, fmt.Errorf("id must be non-empty")
 	}
-	if err = writer.Close(); err != nil {
-		return Response{}, err
+
+	resp, err := c.get(ctx, id)
+	if !resp.IsSuccess() {
+		return api.Response{}, err
 	}
 
-	resp, err := c.client.Create(ctx, body.Bytes(), rest.RequestOptions{
-		ContentType: writer.FormDataContentType(),
-	})
+	part, _ := resp.GetPartWithName("metadata")
+	md, err := unmarshallMetadata(part.Content)
 	if err != nil {
-		return Response{}, err
+		return api.Response{}, err
 	}
 
-	if !rest.IsSuccess(resp) {
-		return Response{}, api.APIError{
-			StatusCode: resp.StatusCode,
-			Body:       body.Bytes(),
-			Request:    rest.RequestInfo{Method: resp.Request.Method, URL: resp.Request.URL.String()},
-		}
+	d := documents.Document{
+		Kind:    string(documentType),
+		Name:    name,
+		Public:  !isPrivate,
```

Comment:
maybe we can change the field name/semantics to `Private`?

Reply:
If we agreed that private is the default, I would keep the name `Public` as in go false is the default value.

In case we when the default for the new document is public, I would go with `Private`